dask/array/random.py

```diff
@@ -360,124 +675,398 @@ def permutation(self, x):
+        return _wrap_func(self, "standard_normal", size=size, chunks=chunks, **kwargs)
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def standard_t(self, df, size=None, chunks="auto", **kwargs):
-        return self._wrap("standard_t", df, size=size, chunks=chunks, **kwargs)
+        return _wrap_func(self, "standard_t", df, size=size, chunks=chunks, **kwargs)
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def tomaxint(self, size=None, chunks="auto", **kwargs):
-        return self._wrap("tomaxint", size=size, chunks=chunks, **kwargs)
+        return _wrap_func(self, "tomaxint", size=size, chunks=chunks, **kwargs)
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def triangular(self, left, mode, right, size=None, chunks="auto", **kwargs):
-        return self._wrap(
-            "triangular", left, mode, right, size=size, chunks=chunks, **kwargs
+        return _wrap_func(
+            self, "triangular", left, mode, right, size=size, chunks=chunks, **kwargs
         )
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def uniform(self, low=0.0, high=1.0, size=None, chunks="auto", **kwargs):
-        return self._wrap("uniform", low, high, size=size, chunks=chunks, **kwargs)
+        return _wrap_func(
+            self, "uniform", low, high, size=size, chunks=chunks, **kwargs
+        )
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def vonmises(self, mu, kappa, size=None, chunks="auto", **kwargs):
-        return self._wrap("vonmises", mu, kappa, size=size, chunks=chunks, **kwargs)
+        return _wrap_func(
+            self, "vonmises", mu, kappa, size=size, chunks=chunks, **kwargs
+        )
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def wald(self, mean, scale, size=None, chunks="auto", **kwargs):
-        return self._wrap("wald", mean, scale, size=size, chunks=chunks, **kwargs)
+        return _wrap_func(self, "wald", mean, scale, size=size, chunks=chunks, **kwargs)
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def weibull(self, a, size=None, chunks="auto", **kwargs):
-        return self._wrap("weibull", a, size=size, chunks=chunks, **kwargs)
+        return _wrap_func(self, "weibull", a, size=size, chunks=chunks, **kwargs)
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def zipf(self, a, size=None, chunks="auto", **kwargs):
-        return self._wrap("zipf", a, size=size, chunks=chunks, **kwargs)
+        return _wrap_func(self, "zipf", a, size=size, chunks=chunks, **kwargs)
 
 
-def _choice(state_data, a, size, replace, p):
-    state = np.random.RandomState(state_data)
-    return state.choice(a, size=size, replace=replace, p=p)
+def _rng_from_bitgen(bitgen):
+    # Assumes typename(bitgen) starts with importable
+    # library name (e.g. "numpy" or "cupy")
+    backend_name = typename(bitgen).split(".")[0]
+    backend_lib = importlib.import_module(backend_name)
```

Comment:
Maybe as a future change we could consider centralizing this check? I think this is the third time I'm seeing the same, maybe a small helper function would reduce code maintenance and be less likely to break by accident.

Reply:
Ah - Thanks for pointing this out. This was one bit of logic that I wasn't super comfortable with in this PR. However, I also found it difficult to come up with a "better way"

Reply:
Unfortunately I don't think there are much better ways, this seems reasonable to me, I'm only proposing we combine this into a single function and call that where necessary. We have something similar (for CuPy types) in https://github.com/dask/dask/blob/60c8d6ea696be3638c595bfe7db1d0f1f3fe4178/dask/utils.py#L1407-L1409, maybe we could add this snippet as a function in that file.